### PR TITLE
Change: Add upgrade voices for Satellite Hack 1 and 2

### PR DESCRIPTION
**Pull request by Stubbjax**
The Satellite Hack upgrades now each have their own EVA audio events, instead of the generic "Upgrade complete" line.

https://user-images.githubusercontent.com/11547761/191956466-2faafe80-274c-4fc0-8873-05f6c67e6168.mp4

### CODE REVIEW

**Comment by ExiLe-ZH** [comment]
Sexy

**Comment by xezon** [comment]
Are the sounds in their respective languages? Also what happens if USA player has Internet Center? Do we expect USA EVA voice as opposed to China EVA voice?

**Comment by Stubbjax** [comment]
> Are the sounds in their respective languages? Also what happens if USA player has Internet Center? Do we expect USA EVA voice as opposed to China EVA voice?

Hmm, that's a good question. How does it work for the other EVA events that @commy2 added in #1026? The localised variants would have to be done by those familiar with those languages to ensure the correct sentence structure, assuming the respective audio exists for them.

**Comment by xezon** [comment]
German, French, Russian can be sourced from locales.

File: Patch104pZH/GameFilesEdited/Data/INI/Voice.ini [hunk 1]
```diff
@@ -5678,3 +5678,18 @@ AudioEvent EvaVoiceUpgradeFortifiedStructure
...
+  Volume = 90
+  Type = ui voice player
+End
+
+AudioEvent EvaVoiceUpgradeSatelliteHackTwo
+  Sounds = ecsat2up
+  Control = random
+  Volume = 90
+  Type = ui voice player
+End
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Voice.ini]:
```suggestion
End

````
Minor nitpick, all lines should end with a /n character,

Comment (Stubbjax) [reply Patch104pZH/GameFilesEdited/Data/INI/Voice.ini]: Ah yeah, I did notice that but forgot about it.

**Review by commy2 — APPROVED** [review]
neat 👍

**Comment by xezon** [comment]
Please do not merge until sounds are sorted.

**Review by commy2 — DISMISSED** [review]
sort out sounds first

**Comment by Stubbjax** [comment]
Aha I wasn't planning to. Just review the PR as `Changes Requested`! (Looks like I was beaten to it.)

**Comment by xezon** [comment]
Additionally, I think we should isolate this change first:
https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1061/commits/71f3bedc8a00510fdc4b2852685f77531a09c566#diff-e8bc6bb844a8f9f70ca66bb955fa3c2cafd736d83bc906c104bc15fdbc942cca

It simplifies how sounds are sourced to avoid English copies. We can extract this commit from #1061 and submit first. If you cannot do this on your own, let me know I will do it.

**Comment by Stubbjax** [comment]
Ah right, didn't see that. It did feel very dirty copying the files, but #1026 was approved and merged that way so I matched the process. Is #1061 awaiting anything in particular?

**Comment by xezon** [comment]
#1061 has not been reviewed. To review, one needs to listen to 1000 sounds. I doubt anyone will do that, so eventually I will just merge it in the future. We can isolate that one commit from #1061, as they do not need to be in one Pull.

**Comment by Stubbjax** [comment]
Alright, so to clarify, cherry-pick commit https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1061/commits/71f3bedc8a00510fdc4b2852685f77531a09c566#diff-e8bc6bb844a8f9f70ca66bb955fa3c2cafd736d83bc906c104bc15fdbc942cca and remove the duplicate audio files from the non-English localisation directories?

**Comment by xezon** [comment]
Yea, that commit can be its own PR. Once that is committed, we do not need to make English copies.

**Comment by Stubbjax** [comment]
> Yea, that commit can be its own PR. 

You will open a dedicated PR?

**Comment by xezon** [comment]
Commit #1273 before this one.

Also regarding English sounds, we should use the cleaned sounds from #1061, not the original ones.

**Comment by commy2** [comment]
lgtm

**Comment by xezon** [comment]
It is still missing German, French, Russian, new English. I can add it if no one beats me to it.

**Comment by xezon** [comment]
> Also what happens if USA player has Internet Center? Do we expect USA EVA voice as opposed to China EVA voice?

I think it is fine if voice is different. When buying Rocket Buggy Ammo as a USA for example then we also hear the GLA Buggy voice.

> The Satellite Hack upgrades now each have their own EVA audio events, instead of the generic "Upgrade complete" line.

Where are these sound files sourced from? I can not find these in original game files.

**Comment by commy2** [comment]
Same place MOAB and Fortified Structures sound if from. It's in a different folder for some reason.

**Comment by xezon** [comment]
I was unable to locate it in all folders. Therefore I also do not know where it is for the other languages. Perhaps it is under a different name? If there is no same audio for German and French then we cannot use this.

**Comment by xezon** [comment]
Here are all the EC files I have access to:

```
ecaactpc.wav
ecabuink.wav
ecabuipc.wav
ecabuiss.wav
ecallreq.wav
ecallyu.wav
ecanklau.wav
ecasslau.wav
ecbasof.wav
ecbasun.wav
ecbeaco.wav
ecblackd.wav
ecbuica.wav
ecbuildg.wav
ecbuilo.wav
ecbuiof.wav
ecbuion.wav
ecbuist.wav
eccapbla.wav
eccapbui.wav
eccapcol.wav
eccapher.wav
eccapjar.wav
eccashs.wav
eccolbud.wav
ecconst.wav
ecdeant.wav
ecdedai.wav
ecdeemp.wav
ecdefue.wav
ecdenuk.wav
ecdepar.wav
ecdescu.wav
eceblacd.wav
ececolbd.wav
ecejarmd.wav
ecempmup.wav
ecfunds.wav
eciobecl.wav
eciocacs.wav
eciocril.wav
eciodecs.wav
eciodesg.wav
eciodess.wav
eciodest.wav
eciofall.wav
ecioproc.wav
eciorise.wav
ecioverl.wav
ecjarmkd.wav
eclagpss.wav
eclanuk.wav
eclapar.wav
eclascu.wav
eclasnea.wav
eclevel.wav
ecneutup.wav
ecnuklau.wav
ecnukrea.wav
econline.wav
ecparact.wav
ecparrea.wav
ecpldef.wav
ecplvic.wav
ecpower.wav
ecrecfnd.wav
ecreinf.wav
ecscdlau.wav
ecscurea.wav
ecselec.wav
ecstrong.wav
ectraing.wav
ectrnfnd.wav
ecunitl.wav
ecunitp.wav
ecunitr.wav
ecupgra.wav
ecvehst.wav
ecyoureq.wav
```

**Comment by xezon** [comment]
I have remade the English sounds. And made French, German ones too. And added default Upgrade sound for Russian. Documentation added. Sounds good.

**Comment by xezon** [comment]
Fixed one small error in French sound.

**Review by xezon — APPROVED** [review]

**Pull request merged**